File: app/src/main/java/com/example/ajay/contacts_4/SyncService.java
package com.example.ajay.contacts_4;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;
import android.support.annotation.Nullable;

/**
 * Created by ajay on 28/9/15.
 */
public class SyncService extends Service {

    private static final Object sSyncAdapterLock = new Object();
    private static SyncAdapter mSyncAdapter = null;

    @Override
    public void onCreate() {
        Log.I("Sync Service created.");
        synchronized (sSyncAdapterLock){
            if(mSyncAdapter == null){
                mSyncAdapter = new SyncAdapter(getApplicationContext(),true);
            }
        }
    }

    @Nullable
    @Override
    public IBinder onBind(Intent intent) {
        Log.I("Sync Service binded.");
        return mSyncAdapter.getSyncAdapterBinder();
    }
}

File: app/src/main/java/com/ajay/synccontacts/services/SyncService.kt
package com.ajay.synccontacts.services

import android.app.Service
import android.content.Intent
import android.os.IBinder
import android.util.Log
import com.ajay.synccontacts.syncing.SyncAdapter

class SyncService : Service() {

    private val TAG: String = javaClass.simpleName
    private var mSyncAdapter: SyncAdapter? = null

    override fun onCreate() {
        super.onCreate()

        Log.i(TAG, "Sync service created")
        synchronized(this) {
            if(mSyncAdapter == null) {
                mSyncAdapter = SyncAdapter(applicationContext, true)
            }
        }
    }

    override fun onBind(intent: Intent): IBinder {
        Log.i(TAG, "Sync service bound")
        return mSyncAdapter!!.syncAdapterBinder
    }
}
